### PR TITLE
Attempt at fixing the tests in master

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

Closes #2842 closes #2935

## 🚨 Test instructions

`yarn test` shouldn't throw errors

### CODE REVIEW

**Comment by mischnic** [comment]
Related: #2842

File: packages/core/integration-tests/test/fs.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ describe('fs', function() {
       assert.equal(output, 'hello');
     });
 
-    it('should inline a file with fs require destructure', async function() {
+    it.skip('should inline a file with fs require destructure', async function() {
```
Comment (DeMoorJasper) [inline packages/core/integration-tests/test/fs.js]: This is related to babel issues afaik, skipped it for now as babel was locked to <7.4 due to bugs.

Comment (mischnic) [reply packages/core/integration-tests/test/fs.js]: > locked to <7.4 due to bugs.

How long will it stay it that way?

Comment (DeMoorJasper) [reply packages/core/integration-tests/test/fs.js]: Not sure Devon locked it, perhaps the bugs have already been fixed

cc @devongovett

Comment (mischnic) [reply packages/core/integration-tests/test/fs.js]: One issue atm with the old babel: `This experimental syntax requires enabling the parser plugin: 'objectRestSpread'` if the deps contain rest spread (which isn't a proposal anymore)

**Comment by DeMoorJasper** [comment]
Everything passes now 🎉

Ready for another review

**Comment by mischnic** [comment]
The tests pass on my machine, but the output is strange:
```

  FSCache
    ✓ should create directory on ensureDirExists (76ms)
(node:9100) UnhandledPromiseRejectionWarning: #<Object>
(node:9100) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:9100) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:9100) UnhandledPromiseRejectionWarning: #<Object>
(node:9100) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
    ✓ should cache resources (81ms)
    ✓ should return null for invalidated resources
    ✓ should remove file on delete (72ms)
    ✓ should remove from invalidated on write (63ms)
```

**Comment by DeMoorJasper** [comment]
@mischnic that's very strange but probably not caused by this PR

File: packages/core/integration-tests/test/fs.js (hunk 1 above)
Comment (devongovett) [reply packages/core/integration-tests/test/fs.js]: I believe babel is fixed now, so we can upgrade it again.

**Comment by DeMoorJasper** [comment]
Opened a new PR based on #2842 with the more up to date babel version

**Pull request closed without merging**